Test/Microsoft.ApplicationInsights.Test/Shared/Extensibility/Implementation/Tracing/HeartbeatTests.cs

```diff
@@ -0,0 +1,357 @@
+            using (var hbeat = new HeartbeatProvider())
+            {
+                hbeat.Initialize(configuration: null);
+            }
+        }
+
+        [TestMethod]
+        public void InitializeHealthHeartbeatTwiceDoesntFail()
+        {
+            using (var hbeat = new HeartbeatProvider())
+            {
+                hbeat.Initialize(configuration: null);
+                hbeat.Initialize(configuration: null);
+            }
+        }
+
+        [TestMethod]
+        public void InitializeHealthHeartbeatDefaultsAreSetProperly()
+        {
+            using (var hbeat = new HeartbeatProvider())
+            {
+                hbeat.Initialize(configuration: null);
+                Assert.IsTrue(hbeat.ExcludedHeartbeatProperties == null || hbeat.ExcludedHeartbeatProperties.Count() == 0);
+                Assert.AreEqual(hbeat.HeartbeatInterval, HeartbeatProvider.DefaultHeartbeatInterval);
+            }
+        }
+
+        [TestMethod]
+        public void InitializeHealthHeartbeatWithNonDefaultInterval()
+        {
+            TimeSpan nonDefaultInterval = TimeSpan.FromMilliseconds(10000);
+
+            using (var hbeat = new HeartbeatProvider() { HeartbeatInterval = nonDefaultInterval })
+            {
+                hbeat.Initialize(configuration: null);
+                Assert.AreEqual(nonDefaultInterval, hbeat.HeartbeatInterval);
+            }
+        }
+
+        [TestMethod]
+        public void InitializeHealthHeartbeatWithZeroIntervalRevertsToDefault()
+        {
+            using (var hbeat = new HeartbeatProvider() { HeartbeatInterval = TimeSpan.FromMilliseconds(0) })
+            {
+                hbeat.Initialize(configuration: null);
+                Assert.AreEqual(hbeat.HeartbeatInterval, HeartbeatProvider.DefaultHeartbeatInterval);
+            }
+        }
+
+        [TestMethod]
+        public void CanExtendHeartbeatPayload()
+        {
+
+            using (var hbeat = new HeartbeatProvider())
+            {
+                hbeat.Initialize(configuration: new TelemetryConfiguration());
+
+                try
+                {
+                    Assert.IsTrue(hbeat.AddHealthProperty("test01", "this is a value", true));
+                }
+                catch (Exception e)
+                {
+                    Assert.Fail(string.Format(CultureInfo.CurrentCulture, "Registration of a heartbeat payload provider throws exception '{0}", e.ToInvariantString()));
+                }
+            }
+        }
+
+        [TestMethod]
+        public void CanSetDelayBetweenHeartbeats()
+        {
+            TimeSpan userSetInterval = TimeSpan.FromMilliseconds(7252.0);
+
+            using (var hbeat = new HeartbeatProvider())
+            {
+                var config = new TelemetryConfiguration(string.Empty, new StubTelemetryChannel());
+                hbeat.Initialize(configuration: config);
+                Assert.AreNotEqual(userSetInterval, hbeat.HeartbeatInterval.TotalMilliseconds);
```

Comment:
what does this test? userSetInterval has no connection with heat yet.

Reply:
Agreed. This test did something before when we had these settings in the Initialize method. Now I'm using this test to ensure that setting the Interval via the DiagnosticsTelemetryModule gets reflected in the HeartbeatProvider. This is more useful and ensures that the value is kept between module construction & initialization.